### PR TITLE
Nanomax classes to match new python3 master

### PR DESCRIPTION
**Pull request by alexbjorling**
Not sure what the release schedule for 0.4 is, but if this could fit in before release, that would be splendid.

### CODE REVIEW

File: ptypy/experiment/nanomax_streaming.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+#                    break
+            except zmq.ZMQError:
+                # no more data available - working around bug in ptypy here
+                if self.latest_det_index_received < self.info.min_frames * parallel.size:
+                    print('have %u detector frames, waiting...' % (self.latest_det_index_received + 1))
+                    time.sleep(.5)
+                else:
+                    break
+
+        print('-------------------------------------')
```
Comment (bjoernenders) [inline ptypy/experiment/nanomax_streaming.py]: Consider using the logging functionality of ptypy instead of print. Print shoudl be used for debugging.

**Review by bjoernenders — APPROVED** [review]
I had a quick look and I only found the repeated use of "print" as something that you might to change to "logger.info" etc. Otherwise, PtyScan classes are the responsibility of the beamlines. So I agree to merge. But it will probably be in the rolling master now.

File: ptypy/experiment/nanomax_streaming.py (hunk 1 above)
Comment (alexbjorling) [reply ptypy/experiment/nanomax_streaming.py]: Busted! You're right, I should use the logger.

**Comment by daurer** [comment]
Was looking through the open PRs and this one looks kind of ready to be merged. @alexbjorling Do you want this merged into master? Only recommendation would be to replace print statements with the logger (as suggested above).

**Comment by alexbjorling** [comment]
Sorry @daurer I missed this update. Hang on, I'll just put in another tiny feature first and then ask to merge.

**Comment by alexbjorling** [comment]
Ok! Please merge and close this request. @daurer @bjoernenders

File: ptypy/experiment/__init__.py.orig [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                   ('.AMO_LCLS', 'AMOScan'),
+                   ('.DiProI_FERMI', 'DiProIFERMIScan'),
+                   ('.optiklabor', 'FliSpecScanMultexp'),
+                   ('.UCL', 'UCLLaserScan'),
+                   ('.nanomax', 'NanomaxStepscanMay2017'),
+                   ('.nanomax', 'NanomaxStepscanNov2016'),
+                   ('.nanomax', 'NanomaxFlyscanJune2017'),
+                   ('.ALS_5321', 'ALS5321Scan'),
+                   ('.Bragg3dSim', 'Bragg3dSimScan')]
+>>>>>>> master
```
Comment (bjoernenders) [inline ptypy/experiment/__init__.py.orig]: Here is a merge conflict.

**Review by bjoernenders — CHANGES_REQUESTED** [review]

File: ptypy/experiment/__init__.py.orig (hunk 2 above)
Comment (bjoernenders) [reply ptypy/experiment/__init__.py.orig]: I actually don't understand this file __init__.py.orig. Did you commit a backup file by accident?

Comment (alexbjorling) [reply ptypy/experiment/__init__.py.orig]: yep sorry, must have backed it up when resolving some earlier conflict

**Review by bjoernenders — APPROVED** [review]

**Comment by bjoernenders** [comment]
I would do a squash merge, since this a topic/side branch. Is that ok with you?

**Comment by alexbjorling** [comment]
> I would do a squash merge, since this a topic/side branch. Is that ok with you?

of course!

**Comment by bjoernenders** [comment]
Merged and topic branch deleted.

**Pull request merged**